### PR TITLE
[Feature] Discovery enhancements

### PR DESCRIPTION
**Pull request by daltondhcp**
**Features added:**
- Support for duplicate subscription and management group displaynames. 
_Will now use ManagementGroupName instead of DisplayName for Management Groups and SubscriptionName (SubscriptionID) for subscriptions._ 
image
_Existing management groups and subscription structure will be **recreated**_
- Aligned environment variable names with action (env:AZOPS_)
- Supports discovery from custom mg root. 
Enabled with by setting AZOPS_SUPPORT_PARTIAL_MG_DISCOVERY to 1. It will by default then do discovery based on the RBAC permissions of the service principal/user.
AZOPS_PARTIAL_MG_DISCOVERY_ROOT can used to manually define discovery root(s). 
- Enabled support for path over 260 characters including check to validate [LongPathsEnabled ](https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file#enable-long-paths-in-windows-10-version-1607-and-later) for full Windows support. 

**Issues fixed**
- Fixes issue with disabled/deleted subscriptions returned by management group api.

### CODE REVIEW

File: src/AzOps.psm1 [hunk 1]
```diff
@@ -365,14 +367,14 @@ class AzOpsScope {
             $mgId = ((($this.scope -split $this.regex_managementgroupExtract) -split '/') | Where-Object { $_ } | Select-Object -First 1)
 
             if ($mgId) {
-                Write-AzOpsLog -Level Debug -Topic "AzOpsScope" -Message "Querying AzOpsAzManagementGroup for Id: $mgId"
-                $mgDisplayName = ($global:AzOpsAzManagementGroup | Where-Object { $_.Name -eq $mgId }).DisplayName
+                Write-AzOpsLog -Level Debug -Topic "pwsh" -Message "Querying Global variable for AzOpsAzManagementGroup"
```
Comment (krowlandson) [inline src/AzOps.psm1:370]: I'm not sure we want the topic back to [pwsh]?

I've also just gone through the code to set every instance of $global to use lowercase, but I think you've accidentaly reverted this during merge conflict resolution?

Comment (daltondhcp) [reply src/AzOps.psm1:370]: Thanks! I’ll change it back - most likely due to merge conflict.

**Pull request merged**